### ISSUE TITLE
Refactoring Commands

### ISSUE THREAD

**Issue by EltonGohJH** [AY2122S1-CS2103T-W08-1/tp#70]
Refactoring commands to make object oriented commands.  
From add n/Elton to person create n:

**Comment by kaixin-hc** [comment]
==Current idea:== 
person, group and subgroup are contained within lists PersonList, GroupList, SubgroupList(which is per group). The lists inherit from interfaces which show its functionality: `CanCreateObjects`, `CanFindObjects`, `CanListObjects`, `CanDeleteObjects`, `CanEditObjects`, `HasNotesObjects`.

==PARSER LOGIC==

1) check first word to assign the proper list class (PersonList, etc) to a variable, `lst` or identify a general command 
2) check second word(list, find, create, delete, edit, note) to determine what command to call.
3) stick all remaining parameters into a string array. EITHER PARSE HERE OR IN A COMMAND SPECIFIC PARSER. 
4) Create the command & return it. May throw an error here if argument

==COMMAND LOGIC==
If it happens to be one of those commands (list, find, create, delete, edit, note, remove)

1) Create new command using the constructor, which will have a form like `new CreateCommand(Creator c, String[] params)`.
(May call command-specific) parser here to parse params
2) Inside the constructor, handle errors
--> params being of the wrong type, depending on the list
3) when you call `command.execute` --> shld execute the correct way for the creator which was given. Inside execute it will call Creator.create or whatever, and the creator method will polymorphically define create.

------ DISCUSSION NOTES
Creator / Listable will be called by the CreateCommand/ListCommand class etc

> Xu Yukun (group 2103T?):
and Creator/Lister/etc will call the correct parsers and create the necessary stuff

> Xu Yukun (group 2103T?):
and do the command execution and calls to the lists

> Xu Yukun (group 2103T?):
so basically Command -> Executor (calls parser) -> List (with correctly created objects)

* we decided not to have interfaces for person / group and subgroup
